Frameworks/CoreGraphics/CGContextCairo.mm

```diff
@@ -1931,3 +1931,29 @@
     UNLOCK_CAIRO();
     return returnValue;
 }
+CGPathRef CGContextCairo::CGContextCopyPath(void) {
+    CGMutablePathRef copyPath = CGPathCreateMutable();
+    cairo_path_t* caPath = cairo_copy_path(_drawContext);
+    cairo_path_data_t* data;
+
+    for (int i = 0; i < caPath->num_data; i += caPath->data[i].header.length) {
+        data = &caPath->data[i];
+        switch (data->header.type) {
```

Comment:
we should add a default here and fail_fast() to guard against potential changes in the future.